### DISCUSSION TITLE
DatabaseCluster: Cluster requires at least 2 subnets, got 1

### DISCUSSION THREAD

**Discussion by maymayman** [aws/aws-cdk#24724] [General]
### Describe the bug

![Untitled](https://user-images.githubusercontent.com/19524958/226513360-a1f2e3e4-1137-4c39-8c57-631607ead911.png)
![Untitled](https://user-images.githubusercontent.com/19524958/226513619-cb704b71-e296-4abe-ae7f-28db0f6ae486.png)
### Expected Behavior
DatabaseCluster can init successful
### Current Behavior
Cluster requires at least 2 subnets, got 1
![Untitled](https://user-images.githubusercontent.com/19524958/226512673-05052458-b889-48b5-b497-176434dd1211.png)
### Reproduction Steps
in First Stack i create a VPC
`this.mainVpc = new Vpc(this, mainVpcName, {
            ipAddresses: IpAddresses.cidr('192.168.0.0/16'),
            vpcName: mainVpcName,
            maxAzs: 2,
            enableDnsHostnames: true,
            enableDnsSupport: true,
            natGateways: 1,
            subnetConfiguration: [
                {
                  name: 'public-subnet-1',
                  subnetType: SubnetType.PUBLIC,
                  cidrMask: 20,
                },
                {
                  name: 'isolated-subnet-1',
                  subnetType: SubnetType.PRIVATE_ISOLATED,
                  cidrMask: 18,
                },
            ],
        });`
        
   and Next stack I use fromLookup to retrieve that VPC
   `const mainVpc: IVpc = Vpc.fromLookup(this, mainVpcName, {
            vpcName: mainVpcName,
            region: props.env?.region,
        });`

and then create a DatabaseCluster like above

### CDK CLI Version

2.64.0

### Node.js Version

16.17

### OS

Window

### Language

Typescript

### Language Version

3.9.7

### Other information

"aws-cdk": "2.64.0",
    "aws-cdk-lib": "2.64.0",
    "constructs": "10.1.246",
![Untitled](https://user-images.githubusercontent.com/19524958/226512898-4baaacd9-90d3-4492-9411-fa1b5eb471b3.png)

**Comment by pahud** [comment]
Let me reproduce the VPC as yours.

**Comment by pahud** [comment]
Hi

If you create your vpc like below, you can list all the public subnet IDs like this:

```ts
      const vpc = new ec2.Vpc(this, 'Vpc', { 
        ipAddresses: ec2.IpAddresses.cidr('192.168.0.0/16'), 
        maxAzs: 2, 
        enableDnsHostnames: true, 
        enableDnsSupport: true, 
        natGateways: 1, 
        subnetConfiguration: [ 
          { name: 'public-subnet-1', subnetType: ec2.SubnetType.PUBLIC, cidrMask: 20, }, 
          { name: 'isolated-subnet-1', subnetType: ec2.SubnetType.PRIVATE_ISOLATED, cidrMask: 18, }, 
        ], 
      });

      const mysubnets = vpc.selectSubnets({ subnetType: ec2.SubnetType.PUBLIC }).subnetIds.join(',')
      new CfnOutput(this, 'SubnetIDs', { value: mysubnets })
```      

In my case, I got the output indicating 2 public subnets selected.

```
Outputs:
demo-dev2.SubnetIDs = subnet-066a7a9f7353d0e53,subnet-03c55f805018bfafb
```

So I would be confident that `{ subnetType: ec2.SubnetType.PUBLIC }` should select 2 existing public subnets and I can pass this to the vpcSubnets required for DatabaseCluster.instanceProps.

Can you check your vpc selection and make sure it turns 2 subnets?

**Comment by JobaDiniz** [comment]
For whoever arrives here with the same issue as me, the problem I had is I set `MaxAz=1`. You need to set `MaxAzs=2` at least.
